client/components/domains/registrant-extra-info/fr-form.jsx

```diff
@@ -83,38 +77,39 @@ class RegistrantExtraInfoForm extends React.PureComponent {
 		const dateOfBirth = ( dobYears && dobMonths && dobDays ) &&
 			[ dobYears, dobMonths, dobDays ].join( '-' );
 
-		if ( dateOfBirth ) {
-			debug( 'Setting dateOfBirth to ' + dateOfBirth +
-				( moment( dateOfBirth, 'YYYY-MM-DD' ).isValid() ? '' : ' (invalid)' ) );
+		if ( ! dateOfBirth ) {
+			return;
 		}
 
-		this.props.onStateChange( {
-			...newState,
-			...( dateOfBirth ? { dateOfBirth } : {} )
-		} );
+		debug( 'Setting dateOfBirth to ' + dateOfBirth +
+			( moment( dateOfBirth, 'YYYY-MM-DD' ).isValid() ? '' : ' (invalid)' ) );
+
+		this.props.updateContactDetailsCache( {
+			registrantExtraInfo: {
+				...this.props.contactDetails.registrantExtraInfo,
+				dateOfBirth
+			} } );
 	}
 
 	newStateFromEvent( event ) {
 		return {
-			...this.props.values,
+			...this.props.contactDetails.registrantExtraInfo,
 			[ event.target.id ]: event.target.value,
 		};
 	}
 
 	handleChangeEvent = ( event ) => {
-		this.props.onStateChange( this.newStateFromEvent( event ) );
-	}
-
-	// We need a deep comparison to check inside props.values
-	shouldComponentUpdate( nextProps ) {
```

Comment:
Taking this out maybe/likely dumb? I've been stripping back while I played with it probably to a fault.